sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/none/NoneDataSource.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.none
+
+import java.util.Optional
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SaveMode
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.sources.DataSourceRegister
+import org.apache.spark.sql.sources.v2._
+import org.apache.spark.sql.sources.v2.writer._
+import org.apache.spark.sql.types.StructType
+
+/**
+ * The datasource can be used in benchmarking because it forces materialization
+ * of each written row of a dataset without converting fields to other types.
+ * This can be used in caching of datasets without additional overhead of an actions.
+ */
+class NoneDataSource extends DataSourceV2 with BatchWriteSupportProvider with DataSourceRegister {
```

Comment:
waiting for https://github.com/apache/spark/pull/23208 ? cc @cloud-fan @gengliangwang 

Also, do we want to have a NULL data source for streaming?

Reply:
For the purpose of the PR, it doesn't matter on which API this datasource is based. I can rewrite it on API v1

> Also, do we want to have a NULL data source for streaming?

 It addresses benchmarking of batch processing. Not sure about streaming. @hvanhovell WDYT?

Reply:
Just in case, I cannot use `null` as the name of the datasource because it is prohibited as package name.